packages/tracing/package.json

```
@@ -80,8 +80,8 @@
     }
   },
   "sideEffects": [
-    "./npm/dist/index.js",
-    "./npm/esm/index.js",
+    "./dist/index.js",
+    "./esm/index.js",
```

We need to keep `./npm/*` as well, not to break `esm`/`cjs` usage here.

```suggestion
    "./dist/index.js",
    "./esm/index.js",
    "./npm/dist/index.js",
    "./npm/esm/index.js",
```

---

Changed Fixes reference to partial to avoid closing that on merge. FWIW with this fix applied the side effects loaded as expected in my webpack environment without the need for the explicit side effects only import.

Re: the suggestion, there is no npm directory in the package, which is why I didn't include it. That prepack logic linked to in the description strips the npm folder from the path.

```
$ ls node_modules/@sentry/tracing                          12:59:52
total 32
drwxr-xr-x   8 kpdecker  staff   256B Apr 19 19:04 ./
drwxr-xr-x  10 kpdecker  staff   320B Apr 19 18:42 ../
-rw-r--r--   1 kpdecker  staff   1.1K Apr 19 19:04 LICENSE
-rw-r--r--   1 kpdecker  staff   4.7K Apr 19 19:04 README.md
drwxr-xr-x  15 kpdecker  staff   480B Apr 19 19:04 build/
drwxr-xr-x  44 kpdecker  staff   1.4K Apr 19 19:04 dist/
drwxr-xr-x  44 kpdecker  staff   1.4K Apr 19 19:04 esm/
-rw-r--r--   1 kpdecker  staff   2.7K Apr 19 19:04 package.json
```

---

@kpdecker can you tell me which SDK version you used for this the file structure output ? The file structure looks like it's pre 6.19.5 - correct? 

Note that in 6.19.5 we slightly changed our package and build directory structure (see #4838). NPM package files are built into `build/npm/(dist|esm)`. However, the `npm` directory does not end up in the NPM package (as described in the PR). IIRC, we updated the side effects paths to `.npm/(dist|esm)/index.js` in #4874 b/c integration tests were failing without it. 

Can you try adding both paths as suggested by @onurtemizkan? Then let's see if this fixes integration tests and your issue.

---

You're right, I forgot about my local revert. 6.19.6 still lacks the npm directory as you note. 

I doubt that I will have time, but would a PR to update the integration tests to match what happens in prod be something that would be considered? Sounds like it would have prevented the regression in 6.19.5.

Appears that CI exec is blocked until @onurtemizkan removes the block.

---

Approved CI - appreciate all the help here!

---

I'm not sure that I'm able to diagnose the ember failure in CI here. `Ember` object not being found seems unrelated to the changes in this PR but I'm not super familiar with the tests in this project :/

---

Our ember tests are unfortunately a bit flaky. That's why they failed